catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1164,6 +1176,69 @@ public CreateResponse create(CreateRequest createRequest)
         return createResponse;
     }
 
+    private void applyAttributeOverridesToMetacardMap(Map<String, String> attributeOverrideMap,
+            Map<String, Metacard> metacardMap) {
+
+        if (MapUtils.isEmpty(attributeOverrideMap) || MapUtils.isEmpty(metacardMap)) {
+            return;
+        }
+
+        attributeOverrideMap.forEach((attributeMapKey, attributeMapValue) -> metacardMap.forEach((metacardMapKey, metacardMapValue) -> {
```

Comment:
[Here](https://gist.github.com/jrnorth/17c3c2fa035f6c9979eed6c30046823b) is an option for doing this with streams. It might be better than using nested `forEach`s, but I'd like to know what others think. Also, you might be able to simplify this by using a `switch` on the `AttributeFormat` as shown in the gist.